src/main/kotlin/io/bazel/kotlin/builder/tasks/klib/KotlinKlibTaskExecutor.kt

```diff
@@ -0,0 +1,87 @@
+package io.bazel.kotlin.builder.tasks.klib
+
+import io.bazel.kotlin.builder.toolchain.CompilationTaskContext
+import io.bazel.kotlin.builder.toolchain.KotlinToolchain
+import io.bazel.kotlin.builder.utils.addAll
+import io.bazel.kotlin.model.KlibCompilationTask
+import java.nio.file.FileSystem
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Collectors
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.io.path.absolute
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import kotlin.io.path.pathString
+
+@Singleton
+class KotlinKlibTaskExecutor
+  @Inject
+  constructor(
+    private val invoker: KotlinToolchain.K2NativeCompilerInvoker,
+  ) {
+    private val fileSystem: FileSystem = FileSystems.getDefault()
+
+    fun execute(
+      context: CompilationTaskContext,
+      task: KlibCompilationTask,
+    ) {
+      task.compile(context)
+    }
+
+    private fun KlibCompilationTask.workingDirectory(): Path = fileSystem.getPath(directories.temp)
+
+    private fun KlibCompilationTask.commonArgs(): MutableList<String> {
+      val workDir = workingDirectory()
+      if (!workDir.exists()) {
+        workDir.toFile().mkdirs()
+      }
+
+      val autoCacheDirectory = workingDirectory().resolve("native_auto_cache")
+      if (!autoCacheDirectory.exists()) {
+        autoCacheDirectory.toFile().mkdirs()
+      }
+
+      val autoCacheFromDirectory = workingDirectory().resolve("native_auto_from")
+      if (!autoCacheFromDirectory.exists()) {
+        autoCacheFromDirectory.toFile().mkdirs()
+      }
+
+      val execRoot = fileSystem.getPath(".").absolute()
+      return mutableListOf<String>().apply {
+        addAll(passThroughFlagsList)
+        // kotlin-native klib compilation requires caching and no way to disable, so pass a temp directory
+        // within the current working directory to isolate its cache (which is unique to this worker anyway)
+        // Ideally we disable caching though and rely only on Bazel
+        add("-Xauto-cache-dir=${autoCacheDirectory.absolutePathString()}")
+        add("-Xauto-cache-from=${autoCacheFromDirectory.absolutePathString()}")
+        add("-Xklib-normalize-absolute-path")
```

Comment:
[normalize-absolute-path](https://github.com/JetBrains/kotlin/blob/7038c0d4c963236ade7861755ba72e1b05a0ca46/compiler/ir/serialization.common/src/org/jetbrains/kotlin/backend/common/serialization/IrSerializationSettings.kt#L29-L31) will only fix the slashes.

In order to support rbe and remote caching, we'll need to do something intelligent with relative paths in klibs](https://kotlinlang.org/docs/native-libraries.html#using-relative-paths-in-klibs).